src/main/java/org/terasology/launcher/gui/javafx/ApplicationController.java

```diff
@@ -504,6 +523,56 @@ private void initComboBoxes() {
         });
     }
 
+    private void selectJobBoxItemForJob(final ComboBox<PackageItem> jobBox, final String jobId) {
+        PackageItem packageItem = jobBox.getItems().stream()
+                .filter(item -> item.versionItems.stream()
+                        .anyMatch(vItem -> vItem.linkedPackageProperty.get().getId().equals(jobId)))
+                .findFirst()
+                .orElseGet(() -> null);
+
+        if (packageItem != null) {
+            jobBox.getSelectionModel().select(packageItem);
+        } else {
+            jobBox.getSelectionModel().select(0);
+        }
+    }
+
+    private void selectBuildVersionBoxItemForVersion(final ComboBox<VersionItem> buildVersionBox, final String version) {
```

Comment:
The implementation of this method then becomes the following, which can probably be inlined wherever needed:

```java
selectFirst(buildVersionBox, item -> item.versionProperty.get().equals(version));
```